src/briefcase/commands/new.py

```diff
@@ -239,49 +276,98 @@ def validate_url(self, candidate):
             raise ValueError("Not a valid URL!")
         return True
 
-    def input_text(self, intro, variable, default, validator=None):
+    def prompt_intro(self, intro: str):
+        """Write the introduction for a prompt."""
+        self.input.prompt()
+        self.input.prompt("-" * MAX_TEXT_WIDTH)
+        self.input.prompt()
+        # textwrap isn't really designed to format text that already contains newlines.
+        # So, instead, break the intro by newlines and format each line individually.
+        self.input.prompt(
+            "\n".join(
+                "\n".join(textwrap.wrap(line, MAX_TEXT_WIDTH))
+                for line in intro.splitlines()
+            )
+        )
+
+    def validate_user_input(self, validator, answer) -> bool:
+        """Validate a user's input is acceptable.
+
+        A warning message is issued if input is enabled. Otherwise, an exception is
+        raised and project creation is aborted.
+        """
+        error_msg = f"Invalid value; {answer}"
+        try:
+            if validator is None or validator(answer):
+                return True
+        except ValueError as e:
+            error_msg = str(e)
+
+        if not self.input.enabled:
+            raise BriefcaseCommandError(error_msg)
+        self.logger.warning()
+        self.logger.warning(error_msg)
+
+        return False
+
+    def input_text(self, intro, variable, default, validator=None, override_value=None):
         """Read a text answer from the user.
 
-        :param intro: An introductory paragraph explaining the question being
-            asked.
+        :param intro: An introductory paragraph explaining the question being asked.
         :param variable: The name of the variable being entered.
-        :param default: The default value if the user hits enter without typing
-            anything.
-        :param validator: (optional) A validator function; accepts a single
-            input (the candidate response), returns True if the answer is
-            valid, or raises ValueError() with a debugging message if the
-            candidate value isn't valid.
-        :returns: a string, guaranteed to meet the validation criteria of
-            ``validator``.
+        :param default: The default value if the user hits enter without typing anything.
+        :param validator: (optional) A validator function; accepts a single input (the
+            candidate response), returns True if the answer is valid, or raises
+            ValueError() with a debugging message if the candidate value isn't valid.
+        :param override_value: value to return instead of soliciting input
+        :returns: a string, guaranteed to meet the validation criteria of ``validator``
         """
-        self.input.prompt(intro)
+        self.prompt_intro(intro=intro)
```

Comment:
Do we need to display the intro prompt if there's an override? Agreed we should display the "Using override" text, but the only reason I can see to display the text is so that in the case of bad input, the context for fixing it is known. I'd argue it's better to just raise an error on the basis that bad command line input can/should be fixed.

Reply:
My thought process here was to only pivot the experience of creating a new project based on whether `--no-input` was provided.

When `--no-input` is absent (i.e. "normal"), everything is printed; however, when it is present, anything written with `input.prompt()` is hidden.

If we try to tweak the "normal" experience based on whether an override value exists, I see two obstacles:
1) Printing the new divider between questions becomes more difficult
    - This divider is a part of `prompt_intro()`....so it'd have to be separated out
    - If we omit the divider for questions with overrides, then I think it's a little confusing because it will appear to be associated with the preceding question
2) If the override is invalid...then we're in a pickle
    - When an override is considered validated, the loop runs again and now solicits the user
    - But we'll need to now show the prompt....this means the prompt printing needs to be moved in to the loop....and we'll need to track already showing it
    - Alternatively, I suppose the approach could be re-implemented like this...
```python
def input_text(self, intro, variable, default, validator=None, override_value=None):
        if override_value:
            if self.validate_user_input(validator, override_value):
                return override_value
        
        self.prompt_intro(intro=intro)

        while True:
            self.input.prompt()

            answer = self.input.text_input(
                prompt=f"{titlecase(variable)} [{default}]: ",
                default=default,
            )

            if self.validate_user_input(validator, answer):
                return answer
```
Given that arguably may be a better implementation anyway...it gives more credence to this idea I suppose.

I would still like to print the divider; so, I can make that it's own method. I would also thinking of incorporating the essence of the question in to the divider.

Maybe something like this:
```
-- Formal Name ----------------------------------------------------------------

First, we need a formal name for your application.

This is the name that will be displayed to humans whenever the name of the
application is displayed. It can have spaces and punctuation if you like, and
any capitalization will be used as you type it.

Formal Name [Hello World]: 

-- Package Name ---------------------------------------------------------------

Next, we need a name that can serve as a machine-readable Python package name
for your application.

This name must be PEP508-compliant - that means the name may only contain
letters, numbers, hyphens and underscores; it can't contain spaces or
punctuation, and it can't start with a hyphen or underscore.

Based on your formal name, we suggest an app name of 'helloworld', but you can
use another name if you want.

App Name [helloworld]: 

-- Bundle Identifier ----------------------------------------------------------

Now we need a bundle identifier for your application. App stores need to
protect against having multiple applications with the same name; the bundle
identifier is the namespace they use to identify applications that come from
you. The bundle identifier is usually the domain name of your company or
project, in reverse order.

For example, if you are writing an application for Example Corp, whose website
is example.com, your bundle would be ``com.example``. The bundle will be
combined with your application's machine readable name to form a complete
application identifier (e.g., com.example.helloworld).

Bundle Identifier [com.example]: 
```
It's a bit repetitive....but it also kinda nicely bookends the question...

Reply:
> My thought process here was to only pivot the experience of creating a new project based on whether `--no-input` was provided.
> 
> When `--no-input` is absent (i.e. "normal"), everything is printed; however, when it is present, anything written with `input.prompt()` is hidden.

That's fair - the use case for using `-Q` and *not* using `--no-input` is admittedly small. 

> If we try to tweak the "normal" experience based on whether an override value exists, I see two obstacles:
> 
> 1. Printing the new divider between questions becomes more difficult
>    
>    * This divider is a part of `prompt_intro()`....so it'd have to be separated out
>    * If we omit the divider for questions with overrides, then I think it's a little confusing because it will appear to be associated with the preceding question

Agreed that there should still be a divider; implementation complexity notwithstanding, I was thinking we'd just avoid giving the impression that a question was actually asked.

> 2. If the override is invalid...then we're in a pickle
>    
>    * When an override is considered validated, the loop runs again and now solicits the user
>    * But we'll need to now show the prompt....this means the prompt printing needs to be moved in to the loop....and we'll need to track already showing it
>    * Alternatively, I suppose the approach could be re-implemented like this...
...
> Given that arguably may be a better implementation anyway...it gives more credence to this idea I suppose.

That was more in line with what I was thinking - that the core "question" loop itself wasn't quite right in the circumstances.
 
> I would still like to print the divider; so, I can make that its own method. I would also thinking of incorporating the essence of the question in to the divider.
...
> It's a bit repetitive....but it also kinda nicely bookends the question...

Agreed. Bonus points if the output in the override case is:
```
-- Package Name ----------------------------------------------------------

Using override value 'org.foobar' for Package Name.

-- Bundle Identifier ----------------------------------------------------------
...
```

The only additional thought I've got in that space is whether it's an opportunity to surface the actual variable name for -Q purposes (.e.g, `--- Formal Name (formal_name) ------`) - but I'm not convinced that's worth it given that it's potentially confusing detail for a new user.

Reply:
This is all in here now; I also tweaked the presentation of the actual generation of the new project. I did have an impulse to add some festive emojis....but I held back...